FirebaseAuth/Sources/AuthProvider/Phone/FIRPhoneAuthProvider.m

```diff
@@ -752,8 +755,32 @@ - (void)reCAPTCHAURLWithEventID:(NSString *)eventID completion:(FIRReCAPTCHAURLC
                                              [NSString stringWithFormat:kReCAPTCHAURLStringFormat,
                                                                         authDomain]];
                                      [components setQueryItems:queryItems];
-                                     if (completion) {
-                                       completion([components URL], nil);
+                                     if (appCheck) {
+                                       [appCheck
+                                           getTokenForcingRefresh:false
+                                                       completion:^(
+                                                           id<FIRAppCheckTokenResultInterop> _Nonnull tokenResult) {
+                                                         if (tokenResult.error) {
+                                                           FIRLogWarning(
+                                                               kFIRLoggerAuth, @"I-AUT000018",
+                                                               @"Error getting App Check token; "
+                                                               @"using placeholder token "
+                                                               @"instead. Error: %@",
+                                                               tokenResult.error);
+                                                         }
+                                                         NSString *appCheckTokenFragments =
+                                                             [@"#fac=" stringByAppendingString:
```

Comment:
You should leave out the "#" here -- since you're explicitly setting this as a fragment field, `NSURLComponents` should handle the structure of the URL for you (you can print out the URL string at the end to verify).